lib/tasks/scripts/clean_biswanath_dupes.rb

```diff
@@ -0,0 +1,223 @@
+class CleanBiswanathDupes
+  def self.call(*args)
+    new(*args).call
+  end
+
+  attr_accessor :dryrun, :exact_matches, :ambiguous_matches, :no_matches
+
+  def initialize(dryrun: false)
+    @dryrun = dryrun
+    @exact_matches = {}
+    @ambiguous_matches = []
+    @no_matches = []
+  end
+
+  def call
+    puts 'Identifying patient matches...'
+    identify_patient_matches
+
+    return if dryrun
+
+    puts 'Porting activity of patients with exact matches...'
+    port_exact_match_activity
+
+    puts 'Porting unmatched patients to the import user...'
+    port_unmatched_patients
+
+    puts 'Deactiviting ambiguous patients...'
+    deactivate_ambiguous_patients
+
+    puts 'Complete. Goodbye.'
+  end
+
+  def identify_patient_matches
+    duplicate_patients.each do |patient|
+      matches = match_by_name_age(patient)
+      if matches.count == 0
+        (no_matches << patient) && next
+      end
+
+      if matches.count == 1
+        (exact_matches[patient.id] = matches.first.id) && next
+      end
+
+      matches = match_by_name_age_address(patient)
```

Comment:
This method should just be `match_by_address`, since the `match_by_name_age` method does the rest.

Reply:
Ah, I see. Unfortunately, we don't re-use the `matches` from the previous method call, we start a fresh search for "by name, age, and address". It's just a design choice we didn't think much about for a one-off task.

Given that this starts a fresh search, do you think that this naming makes sense? Does this also explain your other comment?